File: Day-11/ValidPalindrome.java
public class ValidPalindrome {
    public boolean isPalindrome(String s) {
        if(s.equals(" "))  return true;
        else{
            StringBuilder sb=new StringBuilder();
            for(char c:s.toCharArray())
            {
                if(Character.isLetterOrDigit(c))
                {
                    sb.append(Character.toLowerCase(c));
                }
            }
            String str=sb.toString();
            if(sb.reverse().toString().equals(str))
            {
                return true;
            }
        }
         return false;
    }
}

File: Day-11/ValidPalindrome.py
class ValidPalindrome:
    def isPalindrome(self, s: str) -> bool:
        sb=[]
        for c in s:
            if c.isalnum():
                sb.append(c.lower())
        str="".join(sb)
        rev=str[::-1]
        return str==rev
        